Backend/PostgresConnector.py

```diff
@@ -17,87 +17,111 @@ def __init__(self):
                 password="dad_pass",
                 port = "5432")
 
+    def reconnect(self):
+        self.connection = psycopg2.connect(
+                host="localhost",
+                dbname="dad",
+                user="dad_user",
+                password="dad_pass",
+                port = "5432")
+
     def getAllSystems(self):
         columns = '*'
         sql = "SELECT " + columns + " FROM functions_dim_1_NF"
-        cur = self.connection.cursor()
-        cur.execute(sql)
-        result = cur.fetchall()
-        cur.close()
-        return result
+        try:
+            cur = self.connection.cursor()
+            cur.execute(sql)
+            result = cur.fetchall()
+        except Exception as e:
+            cur.close()
+            return None
+        else:
+            cur.close()
+            return result
 
     # gets a system identified by its label, input is string
     def getSystem(self,label):
         columns = '*'
         sql = "SELECT " + columns + " FROM functions_dim_1_NF WHERE label = '" + label + "'"
         cur = self.connection.cursor()
-        cur.execute(sql)
-        result = cur.fetchall()
-        cur.close()
-        return result
+        try:
```

Comment:
try, except, and finally blocks are good. but in addition, we should use the "with" statement.